.gitlab/ci.sh

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env bash
+
+set -Eeuxo pipefail
+
+source "$CI_PROJECT_DIR/.gitlab/common.sh"
+
+
+export GHCUP_INSTALL_BASE_PREFIX="$CI_PROJECT_DIR/toolchain"
+export CABAL_DIR="$CI_PROJECT_DIR/cabal"
+
+case "$(uname)" in
+    MSYS_*|MINGW*)
+        export CABAL_DIR="$(cygpath -w "$CABAL_DIR")"
+		GHCUP_BINDIR="${GHCUP_INSTALL_BASE_PREFIX}/ghcup/bin"
+        ;;
+	*)
+		GHCUP_BINDIR="${GHCUP_INSTALL_BASE_PREFIX}/.ghcup/bin"
+		;;
+esac
+
+mkdir -p "$CABAL_DIR"
+mkdir -p "$GHCUP_BINDIR"
+export PATH="$GHCUP_BINDIR:$PATH"
+
+export BOOTSTRAP_HASKELL_NONINTERACTIVE=1
+export BOOTSTRAP_HASKELL_GHC_VERSION=$GHC_VERSION
+export BOOTSTRAP_HASKELL_CABAL_VERSION=$CABAL_INSTALL_VERSION
```

Comment:
One minor nitpicking here: Could you wrap these in double quotes?(Though I think these don't contain any spaces and even so this is safe in most modern shells.)

Reply:
It doesn't matter if they contain spaces. This is a variable assignment.

```sh
$ foo="abc def"
$ export bar=$foo
$ echo $bar
abc def
```

Reply:
But I think you're right, some exotic shells exhibit different behavior when `export` is in front of the assignment: https://unix.stackexchange.com/a/97569

Without the `export`, no shell needs the quotes.

Reply:
Yes, some ancient shells require double quotes here, but it's safe for modern shells, so as I said, it's just a nitpicking thing.